src/rule_checker/rule_checker.ts

```diff
@@ -137,11 +137,17 @@ export class RuleChecker implements IRuleChecker {
         }
 
         return (existing: Key): boolean => {
+            const existingPID = existing.split(':')[0];
```

Comment:
Please use the pidFromKey() method which is in attribute_info.ts. There are a number of places in the PR where this comment applies.

Reply:
I see that the original code was written before pidFromKey() and you were just following the pattern. We should fix it any way.

Reply:
Done. 

`pidFromKey()` returns a number and `predicate()` expects a string so I'm using `.toString()` for now. I didn't think it was right to change `pidFromKey()` to return a string. Unless everywhere this is used is in string form?